packages/components/src/shortcuts/ShortcutRegistry.ts

```diff
@@ -1,41 +1,56 @@
+/* eslint-disable max-classes-per-file */
+import { EventTarget } from 'event-target-shim';
+import { Log } from '@deephaven/log';
+import { CustomEventMap, EventShimCustomEvent } from '@deephaven/utils';
 import Shortcut, { KeyState } from './Shortcut';
 
-export default class ShortcutRegistry {
-  static readonly shortcutMap = new Map<string, Shortcut>();
+const log = Log.module('ShortcutRegistry');
 
-  static readonly shortcutsByCategory = new Map<string, Shortcut[]>();
+type EventMap = CustomEventMap<{
+  onUpdate: CustomEvent<Shortcut>;
+}>;
+
+class ShortcutRegistry extends EventTarget<EventMap, 'strict'> {
+  readonly shortcutMap = new Map<string, Shortcut>();
+
+  readonly shortcutsByCategory = new Map<string, Shortcut[]>();
 
   /**
    * Creates a Shortcut and adds it to the registry
    * @param params The constructor params for the {@link Shortcut}
    * @returns The created Shortcut
    */
-  static createAndAdd(
-    params: ConstructorParameters<typeof Shortcut>[0]
-  ): Shortcut {
+  createAndAdd(params: ConstructorParameters<typeof Shortcut>[0]): Shortcut {
     const shortcut = new Shortcut(params);
-    ShortcutRegistry.add(shortcut);
+    this.add(shortcut);
     return shortcut;
   }
 
   /**
    * Adds a shortcut to the registry. Throws if a shortcut with the same ID already exists
    * @param shortcut Shortcut to add to the registry
    */
-  static add(shortcut: Shortcut): void {
-    if (ShortcutRegistry.shortcutMap.has(shortcut.id)) {
-      throw new Error(
+  add(shortcut: Shortcut): void {
+    if (this.shortcutMap.has(shortcut.id)) {
+      log.warn(
```

Comment:
Why was this changed from a throw to a warning?

Reply:
It was getting really annoying in dev b/c hot reload would cause it to throw errors

Also I figured it would be preferable to just have the shortcut potentially broken instead of breaking other things if there's some shortcut registry version inconsistency?

Could change `createAndAdd` to return the shortcut in the registry instead of the created shortcut that isn't added if it's a duplicate ID

Reply:
Warn and return existing shortcut with the same ID